src/org/traccar/protocol/TotemProtocolDecoder.java

```diff
@@ -193,9 +221,10 @@ protected Object decode(
         position.setDeviceId(deviceSession.getDeviceId());
 
         if (pattern == PATTERN1 || pattern == PATTERN2) {
-
-            position.set(Position.KEY_ALARM, parser.next());
-
+            String alarm = decodeAlarm(parser.next());
```

Comment:
You can do something like this instead of adding a variable:
```java
if (parser.hasNext()) {
    position.set(Position.KEY_ALARM, decodeAlarm(parser.next()));
}
```
That's the code style I usually use for optional fields.